rust/src/transformations/make_stable_expr/expr_alias/mod.rs

```diff
@@ -0,0 +1,76 @@
+use std::collections::BTreeSet;
+use std::mem::replace;
+
+use polars_plan::dsl::Expr;
+
+use crate::core::{ExprFunction, Function, MetricSpace, StabilityMap, Transformation};
+use crate::domains::{ExprDomain, OuterMetric};
+use crate::error::*;
+use crate::transformations::DatasetMetric;
+
+use super::StableExpr;
+
+#[cfg(test)]
+mod test;
+
+/// Make a Transformation that renames a column in a LazyFrame.
+///
+/// # Arguments
+/// * `input_domain` - Expr domain
+/// * `input_metric` - The metric under which neighboring LazyFrames are compared
+/// * `expr` - The alias expression
+pub fn make_expr_alias<M: OuterMetric>(
+    input_domain: ExprDomain,
+    input_metric: M,
+    expr: Expr,
+) -> Fallible<Transformation<ExprDomain, ExprDomain, M, M>>
+where
+    M::InnerMetric: DatasetMetric,
+    M::Distance: Clone,
+    (ExprDomain, M): MetricSpace,
+    Expr: StableExpr<M, M>,
+{
+    let Expr::Alias(input, name) = expr else {
+        return fallible!(MakeTransformation, "expected alias expression");
+    };
```

Comment:
This comes up in most of these, but I don't really understand why the type couldn't be tighter

Reply:
The type can only as specific as `Expr`, not a specific variant like `Expr::Alias`.